src/circuit/circuit_frame.rs

```diff
@@ -2246,94 +2191,283 @@ fn apply_continuation<F: PrimeField, CS: ConstraintSystem<F>>(
             )?;
 
             let continuation_is_tail = alloc_equal(
-                &mut cs.namespace(|| "call2 continuation is tail using newer continuation"),
+                &mut cs.namespace(|| "Call2: continuation is tail"),
                 continuation.tag(),
                 &g.tail_cont_tag,
             )?;
 
             let tail_cont = AllocatedContPtr::pick(
-                &mut cs.namespace(|| "call2 the tail continuation using newer continuation"),
+                &mut cs.namespace(|| "Call2: the tail continuation"),
                 &continuation_is_tail,
                 &continuation,
                 &newer_cont,
             );
 
-            (body_form, newer_env, tail_cont)
+            let result_is_fun = alloc_equal(
+                cs.namespace(|| "Call2: result_is_fun"),
+                function.tag(),
+                &g.fun_tag,
+            )?;
+            let args_is_dummy = arg_t.alloc_equal(
+                &mut cs.namespace(|| "Call2: args_is_dummy"),
```

Comment:
Naming is funny here. `args_is_dummy` vs `arg_t`. Args and arg mean something quite different, in this context, right?